sim-cli/src/main.rs

```diff
@@ -182,11 +182,20 @@ async fn main() -> anyhow::Result<()> {
             },
         };
 
+        for validated_activity in &validated_activities {
```

Comment:
Build a hashmap like we do for alias and pubkey rather than iterating through every activity every time